### ISSUE TITLE
Node versions

### ISSUE THREAD

**Issue by strobelm** [CindyJS/CindyJS#562]
Hi we had some problems with node today:

My version of node in Fedora 25 (6.9) had problems with gulp:

> npm ERR! Linux 4.8.15-300.fc25.x86_64
> npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "start"
> npm ERR! node v6.9.2
> npm ERR! npm  v3.10.9
> npm ERR! code ELIFECYCLE
> npm ERR! cindyjs-website@0.0.1 start: `gulp`
> npm ERR! Exit status 1
> npm ERR! 
> npm ERR! Failed at the cindyjs-website@0.0.1 start script 'gulp'.
> npm ERR! Make sure you have the latest version of node.js and npm installed.
> npm ERR! If you do, this is most likely a problem with the cindyjs-website package,
> npm ERR! not with npm itself.
> npm ERR! Tell the author that this fails on your system:
> npm ERR!     gulp
> npm ERR! You can get information on how to open an issue for this project with:
> npm ERR!     npm bugs cindyjs-website
> npm ERR! Or if that isn't available, you can get their info via:
> npm ERR!     npm owner ls cindyjs-website
> npm ERR! There is likely additional logging output above.
> 
> npm ERR! Please include the following file with any support request:
> npm ERR!     /home/michael/Dokumente/promotion/website/npm-debug.log

Furthermore i had to recompile sass:

> Found bindings for the following environments:
>   - Linux 64-bit with Node.js 4.x
> 
> This usually happens because your environment has changed since running `npm install`.
> Run `npm rebuild node-sass` to build the binding for your current environment.

and @richter-gebert node (v. 5.4) would not compile CindyJS at all, saying something is not a TypedArray.

Overall we should perhaps think of always using a node executable we know to be sane. The build script could be adopted to download a certain version even if we have a installed version of node on the system.

**Comment by gagern** [comment]
Hmmm… We do have some provisions fox using certain versions of node: [`make` will run](https://github.com/CindyJS/CindyJS/blob/v0.8.3/Makefile#L39) the [node version check](https://github.com/CindyJS/CindyJS/blob/v0.8.3/make/check-node-version.js) and download a preferined version of node if that check fails. Currently the check tests for broad ranges of versions, though, and [the default version is 0.12.6](https://github.com/CindyJS/CindyJS/blob/v0.8.3/Makefile#L27) which is positively ancient.

Usually I try to use system-installed versions of tools, and try to have my system pretty much on the bleeding edge so that I detect many issues before they hit others. But this time it apparently failed to work that way. As long as there is a way to opt out (e.g. by invoking `node make` directly without going through the Makefile), fixing a version seems reasonable to me. Do we want to manage that ourselves, or use nvm in some way? And what version would you pick?

Since the error you cite seems to be from the webpage, that is yet another story. We don't have any provisions there for switching the version. Nor do we have a Makefile which could handle such things. I'm not sure we want all that overhead, instead of simply asking contributors to remove their `node_modules` directory and start afresh if any such issues turn up. If that doesn't help, it's time for a bug report. Even if we fix the version, we should try to cover other versions as well, particularly if there is a good chance that all future versions will be affected by a given problem.

**Comment by kranich** [comment]
Probably obsolete after #640.
